packages/commonwealth/client/scripts/state/api/topics/createTopic.ts

```diff
@@ -0,0 +1,52 @@
+import axios from 'axios';
+import app from 'state';
+import Topic from 'models/Topic';
+import { useMutation } from '@tanstack/react-query';
+import { ApiEndpoints, queryClient } from 'state/api/config';
+
+interface CreateTopicProps {
+  name: string;
+  description: string;
+  telegram?: string;
+  featuredInSidebar: boolean;
+  featuredInNewPost: boolean;
+  tokenThreshold: string;
+  defaultOffchainTemplate: string;
+}
+
+const createTopic = async ({
+  name,
+  description,
+  telegram,
+  featuredInSidebar,
+  featuredInNewPost,
+  tokenThreshold,
+  defaultOffchainTemplate,
+}: CreateTopicProps) => {
+  const response = await axios.post(`${app.serverUrl()}/createTopic`, {
+    name,
+    description,
+    telegram,
+    featured_in_sidebar: featuredInSidebar,
+    featured_in_new_post: featuredInNewPost,
+    token_threshold: tokenThreshold || '0',
+    default_offchain_template: defaultOffchainTemplate,
+    jwt: app.user.jwt,
+    chain: app.activeChainId(),
+  });
+
+  return new Topic(response.data.result);
+};
+
+const useCreateTopicMutation = () => {
+  return useMutation({
+    mutationFn: createTopic,
+    onSuccess: async (data) => {
+      await queryClient.invalidateQueries({
+        queryKey: [ApiEndpoints.BulkTopics, data.chainId],
+      });
+    },
+  });
+};
```

Comment:
this is the hook that we should export and which use API call function as query or mutation.
Here we can add `onMutate`, `onError` etc. For now it only invalidates query so we fetch fresh topic data right away.